librz/analysis/arch/arm/arm_esil32.c

```diff
@@ -641,27 +641,26 @@ r6,r5,r4,3,sp,[*],12,sp,+=
 					}
 				} else { // No shift
 					rz_strbuf_appendf(&op->esil, "%s,%s,0xffffffff,&,=[%d],%s,%s,+=",
-						REG(0), MEMBASE(1), str_ldr_bytes, REG(2), MEMBASE(1));
+						REG(0), MEMBASE(1), str_ldr_bytes, MEMINDEX(1), MEMBASE(1));
 				}
-			}
-			if (ISREG(1) && str_ldr_bytes == 8) { // e.g. 'strd r2, r3, [r4]', normally should be the only case for ISREG(1).
+			} else if (ISREG(1) && str_ldr_bytes == 8) { // e.g. 'strd r2, r3, [r4]', normally should be the only case for ISREG(1).
 				if (!HASMEMINDEX(2)) {
 					int disp = MEMDISP(2);
 					char sign = disp >= 0 ? '+' : '-';
 					disp = disp >= 0 ? disp : -disp;
-					rz_strbuf_appendf(&op->esil, "%s,%d,%s,%c,0xffffffff,&,=[4],%s,4,%d,+,%s,%c,0xffffffff,&,=[4]",
-						REG(0), disp, MEMBASE(2), sign, REG(1), disp, MEMBASE(2), sign);
-					if (insn->detail->arm.writeback) {
+					rz_strbuf_appendf(&op->esil, "%s,%s,0xffffffff,&,=[4],%s,4,%s,+,0xffffffff,&,=[4]",
+						REG(0), MEMBASE(2), REG(1), MEMBASE(2));
```

Comment:
same here. why?

Reply:
I think it was supposed to be `sign` instead, and just missed one `%s`

Reply:
Because these are post-indexed memory accesses (see `if (ISPOSTINDEX())` above). Memory access happens first at the address in the base register, then at `address + 4` and afterwards (in the `if (writeback)` case) the base register is updated with `Rn = base + disp`.

Same below but with a register and no disponent.

See the docs: https://developer.arm.com/documentation/ddi0406/c/Application-Level-Architecture/Instruction-Details/Alphabetical-list-of-instructions/STRD--register-?lang=en

Please check it again nonetheless.

Reply:
looks like correct. before it was indeed a sum, and the result was the address but now is directly the value the address without the sum.